common/collision/collision_manager.py

```diff
@@ -0,0 +1,58 @@
+import panda3d.core as p3d
+from panda3d.core import NodePath
+from common.collision.collision_object import CollisionObject
+from common.collision.safe_space import SafeSpace
+from common.tiles.tile_controller import create_new_tile, get_collision_shapes
+
+
+class CollisionManager:
```

Comment:
To co tutaj masz to taki trochę builder

nie podoba mi się trochę przekazywanie flag_collider i bullet_factory tak bezpośrednio do konstruktora
bo to nie jest "core" kolizji tylko coś dodatkowego co nasza gra wymaga ale w ogólności nie jest potrzebne
żeby system kolizji działał

dlatego dobrze by było zrobić np -- pierwsza opcja:
set_flag_collider() który ustawi tylko flagę do cTrav/pusher
set_bullet_factory() -- analogicznie

albo druga opcja - tak żeby już całkowicie obiektowo przekombinować ;) ale wsm może się spodobać
zrobić interfejs/protokół tak jak inne w typings.py 
```python
class SupportsCollisionRegistration(Protocol):
  @abstractmethod
  def get_colliders() -> list[CollisionObject]:
    raise NotImplementedException()
  ```
  i wtedy flaga, bulletFactory i playerController będzie implementować to
  a w tym CollisionManagerze
 ```python
 def add_colliders_from(obj: SupportsCollisionRegistration, view=False):
   for collider in obj.get_colliders():
     # ....
 ```
 wtedy konstruktor lżejszy i setup_player_collider niepotrzebny